packages/contracts-bedrock/hardhat.config.ts

```diff
@@ -60,27 +65,275 @@ const config: HardhatUserConfig = {
+      type: 'string', // uint64
+      default: '0x0',
+    },
+    l1GenesisBlockTimestamp: {
+      type: 'number',
+      default: 'changeme', // required for L1 genesis generation, optional for deploy config
+    },
+    // l1GenesisBlockExtraData: not configurable, used for clique singer data. See cliqueSignerAddress instead.
+    cliqueSignerAddress: {
+      type: 'address',
+      default: ethers.constants.AddressZero,
+    },
+    l1GenesisBlockGasLimit: {
+      type: 'string',
+      default: ethers.BigNumber.from(15_000_000).toHexString(),
+    },
+    l1GenesisBlockDifficulty: {
+      type: 'string', // uint256
+      default: '0x1',
+    },
+    l1GenesisBlockMixHash: {
+      type: 'string', // bytes32
+      default: ethers.constants.HashZero,
+    },
+    l1GenesisBlockCoinbase: {
+      type: 'address',
+      default: ethers.constants.AddressZero,
+    },
+    // l1GenesisBlockAlloc: the storage tree is not configurable with deploy-config.
+    l1GenesisBlockNumber: {
+      type: 'string', // uint64
+      default: '0x0',
+    },
+    l1GenesisBlockGasUsed: {
+      type: 'string', // uint64
+      default: '0x0',
+    },
+    l1GenesisBlockParentHash: {
+      type: 'string', // bytes32
+      default: ethers.constants.HashZero,
+    },
+    l1GenesisBlockBaseFeePerGas: {
+      type: 'string', // uint256
+      default: ethers.BigNumber.from(1000_000_000).toHexString(), // 1 gwei
+    },
+
+    // Optional L2 genesis block values. These must ONLY be used by the L2 genesis config script.
+    // -------------------------------------------------
+    l2GenesisBlockNonce: {
+      type: 'string', // uint64
+      default: '0x0',
+    },
+    // l2GenesisBlockTimestamp: configured dynamically, based on the timestamp of l1StartingBlockTag.
+    l2GenesisBlockExtraData: {
+      type: 'string', // important: in the case of L2, which uses post-Merge Ethereum rules, this must be <= 32 bytes.
+      default: ethers.constants.HashZero,
+    },
+    l2GenesisBlockGasLimit: {
       type: 'string',
+      default: ethers.BigNumber.from(15_000_000).toHexString(),
+    },
+    l2GenesisBlockDifficulty: {
+      type: 'string', // uint256
+      default: '0x1',
+    },
+    l2GenesisBlockMixHash: {
+      type: 'string', // bytes32
+      default: ethers.constants.HashZero,
+    },
+    l2GenesisBlockCoinbase: {
+      type: 'address',
+      default: ethers.constants.AddressZero,
+    },
+    // l2GenesisBlockAlloc: the storage tree is not configurable with deploy-config.
+    l2GenesisBlockNumber: {
+      type: 'string', // uint64
+      default: '0x0',
+    },
+    l2GenesisBlockGasUsed: {
+      type: 'string', // uint64
+      default: '0x0',
+    },
+    l2GenesisBlockParentHash: {
+      type: 'string', // bytes32
+      default: ethers.constants.HashZero,
+    },
+    l2GenesisBlockBaseFeePerGas: {
+      type: 'string', // uint256
+      default: ethers.BigNumber.from(1000_000_000).toHexString(), // 1 gwei
```

Comment:
https://github.com/ethers-io/ethers.js/blob/ce8f1e4015c0f27bf178238770b1325136e3351a/packages/units/src.ts/index.ts#L64

`ethers.utils.formatUnits(1, 'gwei')`

Reply:
geth and some others are flexible with changing between hex and decimal values for uint64 strings, but I don't know if every client is. Sticking to hex formatting here.

Reply:
I meant to populate `BigNumber.from` to ensure that the math is correct